src/calculation/ClauseResultsBuilder.ts

```diff
@@ -504,28 +522,63 @@ export function prettyFHIRObject(
+  } else if (resourceType === 'Duration') {
+    return `${typeStr}${result.get('value')?.['value']} ${result.get('unit')?.['value']}`;
+  } else if (resourceType === 'Ratio') {
+    const numer = result.get('numerator')?.['value']?.['value'];
+    const numerUnit = result.get('numerator')?.['unit']?.['value'];
+    const denom = result.get('denominator')?.['value']?.['value'];
+    const denomUnit = result.get('denominator')?.['unit']?.['value'];
+    return `${typeStr}(${numer}${numerUnit ? ` ${numerUnit}` : ''})/(${denom}${denomUnit ? ` ${denomUnit}` : ''})`;
   }
 
-  // Handle resources
-  let prettyResultReturn = '';
-  const currentIndentation = Array(indentLevel).join(' ');
-  prettyResultReturn = prettyResultReturn.concat(`${resourceType.toUpperCase()}\n`);
-  prettyResultReturn = prettyResultReturn.concat(`${currentIndentation}${keyIndentation}ID: ${result.getId()}\n`);
-  if (clinicalDateParams?.base.includes(resourceType)) {
-    const expressionMatch = clinicalDateParams.expression.split('|').find(e => e.trim().startsWith(resourceType));
-    const expression = expressionMatch?.split('.')[1].trim() || '';
+  if (resourceType in parsedCodePaths) {
+    // Handle resources
+    let prettyResultReturn = '';
+    const currentIndentation = Array(indentLevel).join(' ');
+    prettyResultReturn = prettyResultReturn.concat(`${resourceType}\n`);
+    prettyResultReturn = prettyResultReturn.concat(`${currentIndentation}${keyIndentation}ID: ${result.getId()}\n`);
+    if (clinicalDateParams?.base.includes(resourceType)) {
+      const expressionMatch = clinicalDateParams.expression.split('|').find(e => e.trim().startsWith(resourceType));
+      const expression = expressionMatch?.split('.')[1].trim() || '';
+      prettyResultReturn = prettyResultReturn.concat(
+        `${currentIndentation}${keyIndentation}${expression.toUpperCase()}: ${prettyResult(
+          result.getDateOrInterval(expression),
+          false
+        )}\n`
+      );
+    }
+    const codePath = parsedCodePaths[resourceType].primaryCodePath;
     prettyResultReturn = prettyResultReturn.concat(
-      `${currentIndentation}${keyIndentation}${expression.toUpperCase()}: ${prettyResult(
-        result.getDateOrInterval(expression),
-        false
-      )}\n`
+      `${currentIndentation}${keyIndentation}${codePath.toUpperCase()}: ${prettyResult(
+        result.getCode(codePath),
+        false,
+        indentLevel,
+        (keyIndent ?? 0) + codePath.length + 2 // key length + 2 for colon, space
+      )}`
     );
+    return prettyResultReturn;
+  }
+
+  // Handle values
+  if (result.getTypeInfo().findElement('value', true)) {
+    const value = result.get('value');
+
+    if (typeof value === 'string' || typeof value === 'boolean' || typeof value === 'number') {
+      // Try simple value
+      return `${typeStr}${value}`;
+    } else if (
+      typeof value?.['value'] === 'string' ||
```

Comment:
Not sure if this is a big deal but I thought I would point it out to get thoughts on it...but in Matt's bundle, one of the statement results is of resourceType `EncounterStatus` and its value is a string. Since `EncounterStatus` isn't in `parsedCodePaths`, then the pretty result is `ENCOUNTERSTATUS: final`. Do we want the typeStr to be all uppercase in this scenario?

Reply:
Yeaaaah- I did notice this. I think there's going to be a lot of potential variability in what kind of thing ends up being printed from this statement. I would lean toward keeping this all caps to match the rest of the output... or changing everything to not be all caps. The resource type being Pascal case kind of makes sense as a special case, but I don't think this one really does, IMO.